### ISSUE TITLE
:GoAddTags with nested structs

### ISSUE THREAD

**Issue by dvcrn** [fatih/vim-go#990]
Thanks a lot for `:GoAddTags`. It already saved me so much time!

I noticed that tag generation for nested structs seem to not work correctly:

```
type Foo struct {
    One   string
    Two   string
    Three string
    Bar   struct {
        Four string
        Five string
    }
}
```

after :GoAddTags

```
type Foo struct {
    One   string `json:"one"`
    Two   string `json:"two"`
    Three string `json:"three"`
    Bar   struct { `json:"bar"`   <------
        Four string `json:"four"`
        Five string `json:"five"`
    }
}
```

How it should be:

```

type Foo struct {
    One   string `json:"one"`
    Two   string `json:"two"`
    Three string `json:"three"`
    Bar   struct {
        Four string `json:"four"`
        Five string `json:"five"`
    } `json:"bar"`         <------
}
```

**Comment by fatih** [comment]
Thanks @dvcrn 

I'm planning to move `:GoAddTags` to use the tool motion (which supports AST based output). It'll cover all these cases for us. Thanks!

**Comment by dvcrn** [comment]
Hi fatih, is there anything new on this? 
If you could give me some implementation details I could also try to help you with this

**Comment by fatih** [comment]
@dvcrn I'm currently working on this, will update and close this issue once it's ready

**Comment by fatih** [comment]
Fixed with #1024 please, feel free to test and reopen if it's not the case.
